src/lua/datetime.lua

```diff
@@ -913,16 +913,20 @@ local function datetime_parse_from(str, obj)
         check_str(tzname, 'datetime.parse()')
     end
 
+    local date, len
     if not fmt or fmt == '' or fmt == 'iso8601' or fmt == 'rfc3339' then
-        local date, len = datetime_parse_full(str)
-        -- Override timezone.
-        if date.tz == '' and date.tzoffset == 0 then
-            datetime_set(date, obj or {})
-        end
-        return date, tonumber(len)
+        date, len = datetime_parse_full(str)
     else
-        return datetime_parse_format(str, fmt)
+        date, len = datetime_parse_format(str, fmt)
+    end
+
+    -- Override timezone, if it was not specified in a parsed
```

Comment:
It would be nice to add a test to check this `else` branch.

Reply:
The tests already exists:

- Suite "Simple tests for parser" https://github.com/tarantool/tarantool/blob/6fd4544740150d11a6120c8cf7ef7df949527dc9/test/app-tap/datetime.test.lua#L414
- Suite "Check parsing of full supported years range" https://github.com/tarantool/tarantool/blob/6fd4544740150d11a6120c8cf7ef7df949527dc9/test/app-tap/datetime.test.lua#L533C12-L533C55
- Suite "Datetime string parsing by format (detailed)" https://github.com/tarantool/tarantool/blob/6fd4544740150d11a6120c8cf7ef7df949527dc9/test/app-tap/datetime.test.lua#L922C12-L922C56
- Suite "Parse strptime format" https://github.com/tarantool/tarantool/blob/6fd4544740150d11a6120c8cf7ef7df949527dc9/test/app-tap/datetime.test.lua#L1791C12-L1791C33
- I'll add more in https://github.com/tarantool/tarantool/pull/10499

Reply:
My bad, it wasn't obviour for me. Thanks for the clarification!